sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs

```diff
@@ -91,7 +91,7 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
                     else
                     {
                         // Rethrow a singular exception
-                        if (exceptions?.Count == 1)
+                        if (exceptions!.Count == 1)
```

Comment:
why is it necessary here to reinforce that exceptions is not null?

Reply:
Maybe I am missing where nullable is enabled...

Reply:
The compiler has some level of data flow analysis and can tell when the variable was initialized in simple cases but this case is not that simple.

`exceptions` and `lastException` variables are initialized together so when `lastException` is not null the `exceptions` is not-null as well.

https://github.com/azure/azure-sdk-for-net/blob/fefef2a5cc8f17ea3e2472cd2492378ea6f57d1c/sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs#L66-L73 

we check that `lastException` is not null here https://github.com/azure/azure-sdk-for-net/blob/fefef2a5cc8f17ea3e2472cd2492378ea6f57d1c/sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs#L85

and know that `exceptions` would be initialized as well but the compiler doensn't track cross variable dependencies and can't come to the same conclusion.